colin-api/src/colin_api/models/filing.py

```diff
@@ -1414,6 +1417,15 @@ def add_filing(cls, con, filing: Filing) -> int:
             current_app.logger.error(err.with_traceback(None))
             raise err
 
+    @classmethod
+    def is_business_in_lear(cls, bc_identifier: str) -> bool:
+        """Check if business is in lear."""
+        response = LegalApiService.query_business(bc_identifier)
+
+        if response.status_code == HTTPStatus.OK:
+            return True
+        return False
```

Comment:
if it's a 404, i understand returning false but if legal api is returning a 500 or something, i don't think we should be just assuming the business is not in LEAR?

Reply:
If the response is a 500 (or anything other than 200 or 404), can this flow be aborted and it will automatically retry later?

Otherwise, what should be done if this isn't a 200 or 404?

Reply:
i think we can throw an exception in the `LegalApiService` if get business endpoint doesn't return a 404 or 200. 
 And if get business endpoint results in an exception we also re-throw the exception.

If we do this then, for non-404 and non-200 scenarios, it will just bubble up to the exception handler for the add filings endpoint.

https://github.com/bcgov/lear/blob/0502c9f58eb736822485e937001760ca26c4dc21/colin-api/src/colin_api/resources/filing.py#L196-L202

Reply:
So you'd see an error in the logs and this filing wouldn't get processed (synced)?

Reply:
yes, the filing wouldn't get synced if we can't determine whether the business in LEAR.  Either because get business endpoint is broken or legal api is down

Reply:
Sounds good to me.